libr/io/io.c

```diff
@@ -51,69 +52,75 @@ static int al_fd_write_at_wrap (RIO *io, int fd, ut64 addr, ut8 *buf, int len, R
 	} else {
 		log->allocation_failed = true;
 	}
-	return rlen;
 }
 
-typedef int (*cbOnIterMap)(RIO *io, int fd, ut64 addr, ut8 *buf, int len, RIOMap *map, void *user);
+#define	r_io_map_from(x)	((x)->itv.addr)
+#define	r_io_map_to(x)		((x)->itv.addr + (x)->itv.size - 1)
+
+typedef void (*cbOnIterMap)(RIO *io, int fd, ut64 addr, ut8 *buf, int len, RIOMap *map, void *user);
 
 // Precondition: len > 0
 // Non-stop IO
-// Returns true iff all reads/writes on overlapped maps are complete.
-static bool onIterMap(SdbListIter *iter, RIO *io, ut64 vaddr, ut8 *buf,
-		int len, int match_flg, cbOnIterMap op, void *user) {
-	// vendaddr may be 0 to denote 2**64
-	ut64 vendaddr = vaddr + len, len1;
-	int t;
-	bool ret = true;
-	for (; iter; iter = iter->p) {
-		RIOMap *map = (RIOMap *)iter->data;
-		ut64 to = r_itv_end (map->itv);
-		if (r_itv_overlap2 (map->itv, vaddr, len)) {
-			if ((map->flags & match_flg) == match_flg || io->p_cache) {
-				t = vaddr < map->itv.addr
-						? op (io, map->fd, map->delta, buf + map->itv.addr - vaddr,
-								len1 = R_MIN (vendaddr - map->itv.addr, map->itv.size), map, user)
-						: op (io, map->fd, map->delta + vaddr - map->itv.addr, buf,
-								len1 = R_MIN (to - vaddr, len), map, user);
-				if (t != len1) {
-					ret = false;
-				}
+static void onIterMap(SdbListIter* iter, RIO* io, ut64 vaddr, ut8* buf,
```

Comment:
Too many `if` conditions for interval overlapping checks, unnecessary. Your revert is a regression.